src/pybind/mgr/dashboard/controllers/nfsganesha.py

```diff
@@ -285,16 +285,16 @@ def lsdir(self, root_dir=None, depth=1):
             logger.warning("[NFS] Limiting depth to maximum value of 5: "
                            "input depth=%s", depth)
             depth = 5
-        root_dir = '{}/'.format(root_dir) \
-                   if not root_dir.endswith('/') else root_dir
-
+        root_dir = '{}{}'.format(root_dir.rstrip('/'), '/')
         try:
             cfs = CephFS()
+            root_dir = root_dir.encode()
             paths = cfs.get_dir_list(root_dir, depth)
-            paths = [p[:-1] for p in paths if p != root_dir]
-            return {'paths': paths}
+            # Convert (bytes => string) and prettify paths (strip slashes).
+            paths = [p.decode().rstrip('/') for p in paths if p != root_dir.encode()]
```

Comment:
I think the call to `root_dir.encode()` might rise an exception here because you already encoded the `root_dir` string in line 291.